database/config.go

```diff
@@ -134,46 +139,112 @@ func (f *FieldConstraint) ScanDocument(d document.Document) error {
 	return err
 }
 
-type tableConfigStore struct {
-	st engine.Store
+type tableInfo struct {
+	// storeID is a generated ID that acts as a key to reference this table.
+	// The first-4 bytes represents the timestamp in second and the last-2 bytes are
+	// randomly generated.
+	storeID [6]byte
+
+	cfg *TableConfig
```

Comment:
Why do we need a separate `TableConfig` type? What about just keeping everything in the same structure?

```go
type TableInfo struct {
  StoreID [6]byte
  FieldConstraints []FieldConstraint
  LastKey int64
}
```

Reply:
This is a good point. There is no strict reason behind this. I'll do that 👍

Reply:
After reflexion, this change will imply a loooot of diff. I suggest to merge this one, and I'll do that later on in a dedicated PR.